packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -728,7 +831,33 @@ class TextSelectionOverlay {
     _handleSelectionHandleChanged(newSelection, isEnd: false);
   }
 
-  void _handleAnyDragEnd(DragEndDetails details) => _selectionOverlay.hideMagnifier(shouldShowToolbar: !_selection.isCollapsed);
+  void _handleAnyDragEnd(DragEndDetails details) {
+    if (selectionControls is! TextSelectionHandleControls) {
+      _selectionOverlay.hideMagnifier(
+        shouldShowToolbar: !_selection.isCollapsed,
+      );
+      return;
+    }
+    final RenderBox renderBox = context.findRenderObject()! as RenderBox;
```

Comment:
The context may have already been unmounted since this is on a different OverlayEntry.